### PR TITLE
Attach the go service boot logic

### PR DESCRIPTION
**Pull request by omer9564**

### CODE REVIEW

File: horizon/config.py [hunk 1]
```diff
@@ -17,6 +17,36 @@ def __new__(cls, prefix=None, is_model=True):
...
+        None,
+        description="URL to the Go opal service that manages this PDP, typically Permit.io cloud (api.permit.io)",
+    )
+
+    GOPAL_TOKEN = confi.str(
+        "GOPAL_TOKEN",
+        "secret",
+        description="the token to authenticate with the GOPAL service",
+    )
+
```
Comment (asafc) [inline horizon/config.py:49]: really don't like the gopal prefix, not sure what else we can do for now as the logic is split between python and go. see if you can come up with something better.

File: horizon/config.py [hunk 2]
```diff
@@ -17,6 +17,36 @@ def __new__(cls, prefix=None, is_model=True):
...
+    ENABLE_GOPAL = confi.bool(
+        "ENABLE_GOPAL",
+        False,
+        description="if true, the sidecar will enable the GOPAL service to manage the PDP",
+    )
+
+    GOPAL_SERVICE_URL = confi.str(
+        "GOPAL_SERVICE_URL",
+        "http://localhost:8080",
+        description="URL to the Go opal service that manages this PDP, typically Permit.io cloud (api.permit.io)",
```
Comment (asafc) [inline horizon/config.py:29]: "typically Permit.io cloud (api.permit.io)" is incorrect here, this is a local service

File: horizon/config.py [hunk 3]
```diff
@@ -17,6 +17,36 @@ def __new__(cls, prefix=None, is_model=True):
...
+    )
+
+    GOPAL_ENABLE_REMOTE_BACKUP = confi.bool(
+        "GOPAL_ENABLE_REMOTE_BACKUP",
+        False,
+        description="if true, the sidecar will enable the GOPAL service to manage the PDP",
```
Comment (asafc) [inline horizon/config.py:35]: description is incorrect, also - backups should always be enabled, it's part of the solution.
why do we need this config?

File: horizon/config.py [hunk 4]
```diff
@@ -17,6 +17,36 @@ def __new__(cls, prefix=None, is_model=True):
...
+    )
+
+    GOPAL_REMOTE_BACKUP_URL = confi.str(
+        "GOPAL_REMOTE_BACKUP_URL",
+        None,
+        description="URL to the Go opal service that manages this PDP, typically Permit.io cloud (api.permit.io)",
```
Comment (asafc) [inline horizon/config.py:41]: description is incorrect, this is a different service than api.permit.io - it's the (new) backup service.

File: horizon/gopal/runner.py [hunk 5]
```diff
@@ -31,6 +35,7 @@ def _client(self) -> aiohttp.ClientSession:
         return self.__client
 
     async def __aenter__(self):
+        self.set_envs()
```
Comment (asafc) [inline horizon/gopal/runner.py:38]: really don't like the fact we are doing it with env vars.
we probably need to switch to yaml later.

**Review by asafc — APPROVED** [review]
overall LGTM, minor comments

File: horizon/gopal/runner.py (hunk 5 above)
Comment (omer9564) [reply horizon/gopal/runner.py:38]: we will switch to cobra & viper later on so everything will be available but IMO in the beginning this isn't the most important part

File: horizon/config.py (hunk 1 above)
Comment (omer9564) [reply horizon/config.py:49]: changed

File: horizon/config.py (hunk 2 above)
Comment (omer9564) [reply horizon/config.py:29]: fixed

File: horizon/config.py (hunk 3 above)
Comment (omer9564) [reply horizon/config.py]: fixed

File: horizon/config.py (hunk 4 above)
Comment (omer9564) [reply horizon/config.py]: fixed

**Pull request merged**